tlf/handle.go

```diff
@@ -326,3 +338,71 @@ func (h Handle) IsFinal() bool {
 func (h Handle) IsConflict() bool {
 	return h.ConflictInfo != nil
 }
+
+// DeepEqual returns true if the handle is equal to another handle.
+func (h Handle) DeepEqual(other Handle) bool {
+	if len(h.Writers) != len(other.Writers) {
+		return false
+	}
+	if len(h.UnresolvedWriters) != len(other.UnresolvedWriters) {
+		return false
+	}
+	if len(h.Readers) != len(other.Readers) {
+		return false
+	}
+	if len(h.UnresolvedReaders) != len(other.UnresolvedReaders) {
+		return false
+	}
+
+	if !h.sorted {
+		sort.Sort(UIDList(h.Writers))
+		sort.Sort(UIDList(h.Readers))
+		sort.Sort(SocialAssertionList(h.UnresolvedWriters))
+		sort.Sort(SocialAssertionList(h.UnresolvedReaders))
+		h.sorted = true
```

Comment:
This doesn't have a pointer receiver, so isn't setting this caching field pretty useless?

Reply:
Excellent point. Will fix.